lib/api/model/model.dart

```diff
@@ -656,6 +655,38 @@ enum MessageFlag {
   String toJson() => _$MessageFlagEnumMap[this]!;
 }
 
+/// The name of a Zulip topic.
+// TODO(dart): Can we forbid calling Object members on this extension type?
+//   (The lack of "implements Object" ought to do that, but doesn't.)
+//   In particular an interpolation "foo > $topic" is a bug we'd like to catch.
+// TODO(dart): Can we forbid using this extension type as a key in a Map?
+//   (The lack of "implements Object" arguably should do that, but doesn't.)
+//   Using as a Map key is almost certainly a bug because it won't case-fold;
+//   see for example #739, #980, #1205.
```

Comment:
Hmm, yeah. One way this PR is helpful is by making it easy to find topic-keyed maps, where we're being incorrectly case-sensitive 🙂 with `git grep 'Map<TopicName,'`

- `Unreads.streams` (tracked as #980)
- `ChannelStoreImpl.topicVisibility` (no issue yet!)
- `RecentSenders.topicSenders` (no issue yet!)

Do you know if we can make a nice drop-in replacement for `Map<TopicName, …>` that uses the canonicalized form of `TopicName`?

Reply:
Yeah, that is handy.

> Do you know if we can make a nice drop-in replacement for `Map<TopicName, …>` that uses the canonicalized form of `TopicName`?

Yeah. I think we'll want a class with a name like `TopicMap`, which maybe even implements `Map<TopicName, …>`. Its implementation will resemble web's `FoldDict`:
https://github.com/zulip/zulip/blob/e963d5eaa9d73faa31b9c9e1463a2ac0a5169e7a/web/src/fold_dict.ts